vendor/bat-native-ads/src/bat/ads/internal/ml/data/text_data.cc

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <string>
+
+#include "bat/ads/internal/ml/data/text_data.h"
+
+namespace ads {
+namespace ml {
+namespace data {
+
+TextData::TextData() : Data(DataType::TEXT_DATA) {}
+
+TextData::TextData(const TextData& text_data) : Data(DataType::TEXT_DATA) {
+  text_ = text_data.GetText();
+}
+
+TextData& TextData::operator=(const TextData& text_data) {
```

Comment:
See https://google.github.io/styleguide/cppguide.html#Copyable_Movable_Types

Reply:
There is a check on line 44, that prevents such case.

Reply:
? Was this comment meant for here as I do not see line 44?

Reply:
Sorry, wrong place for the comment :(

Reply:
After the short investigation I've remembered why do we need explicit copy assignment operator. Because we have a derived `const DataType type_` member which cannot be copied by default, but there is no need it should remain the same for all instances of the same class. It's used for downcasting by the way.

Reply:
Please leave comments in the code explaining, thanks